capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -100,8 +127,34 @@ export class SearchUserModal extends Component {
     // Owner cannot be a Collaborator
     collaborators = collaborators.filter((collaborator) => collaborator.id !== this.props.bookList.userID);
 
-    this.setState({ addedUsersTracker: collaborators, addedUsers: collaborators });
+    this.setState({ addedUsers: collaborators });
+  }
+
+  fetchMembers = async () => {
 
+    if (!(this.props.type === 'clubs' && this.props.club)) {
+      return;
+    }
+
+    else if (this.props.club.memberIDs.length === 0) {
+      return;
+    }
+
+    let members = [];
+
+    await Promise.all(this.props.club.memberIDs.map(async (memberId) => {
+
+      const member = await fetch(`/api/user?id=${memberId}`).then(resp => resp.json());
```

Comment:
Remove the `async` and `await` here, since that will cause the API requests to go out serially instead of in parallel. The `await Promise.all` above will ensure you don't move on until all the requests have received responses as long as you `return` the result of `fetch` from inside your `map` callback. You can `delete` the `member.tokenObj` inside a `.then` here (or synchronously `map` over the members and do so later), and you can avoid the `push` call if you store the result of `await Promise.all` in `members`.

This is a little tricky: let me know if you'd like to follow up and talk it through together at some point.

Reply:
Oh okay, I think I implemented the changes correctly. I will try to change all the Promise.All in the project to speed up the web app.